### PR TITLE
ShoppingCart implementation

### PR DESCRIPTION
**Pull request by tylerastraub**
added ShoppingCart object to REST API with minimal functionality

### CODE REVIEW

File: ecommerce/src/test/java/com/yetanotheruseless/industrycrashcourse/ecommerce/EcommerceApiTests.java [hunk 1]
```diff
@@ -20,7 +22,7 @@
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @ContextConfiguration(classes=EcommerceApplication.class)
-@WebMvcTest(ProductController.class)
+@WebMvcTest
```
Comment (lkeitadk) [inline ecommerce/src/test/java/com/yetanotheruseless/industrycrashcourse/ecommerce/EcommerceApiTests.java:25]: I'm curious, why do you need to do this?

**Review by lkeitadk — COMMENTED** [review]
I've spent a really long time staring at your code, but I don't really know why it's failing tests. The code makes sense to me and I just can't see where it's going wrong. Perhaps we could go over it together on discord or something

File: ecommerce/src/test/java/com/yetanotheruseless/industrycrashcourse/ecommerce/EcommerceApiTests.java (hunk 1 above)
Comment (tylerastraub) [reply ecommerce/src/test/java/com/yetanotheruseless/industrycrashcourse/ecommerce/EcommerceApiTests.java:25]: i'm not sure, @jakemannix might have a better answer

**Review by lkeitadk — APPROVED** [review]
looks good to me, well done

File: ecommerce/src/main/java/com/yetanotheruseless/industrycrashcourse/ecommerce/shoppingCart/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    @GetMapping(path = "{id}")
+    public ShoppingCart get(@PathVariable Long id) {
+        return shoppingCartService.findById(id).orElseThrow(() -> new RuntimeException("Shopping cart not found. Id: " + id));
+    }
+
+    @PostMapping
+    public Long addShoppingCart(@RequestBody ShoppingCart shoppingCart) {
+//        if (shoppingCart.getName() == null) {
+//            // shopping cart does not have name qualifier... yet
+//        }
```
Comment (jakemannix) [inline ecommerce/src/main/java/com/yetanotheruseless/industrycrashcourse/ecommerce/shoppingCart/ShoppingCartController.java:34]: should remove in your next PR

File: ecommerce/src/main/java/com/yetanotheruseless/industrycrashcourse/ecommerce/shoppingCart/ShoppingCartRepository.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.Optional;
+
+@Repository
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+
+    Optional<ShoppingCart> findOneById(Long id);
+
+    // spring boot does not natively support this type of query, so we have to handwrite it.
+    // convert java.time.Instant to a java.sql.Timestamp using Timestamp.from(Instant)
+    @Query("select cart from ShoppingCart cart where cart.lastUpdatedAtTimestamp <= :instant")
```
Comment (jakemannix) [inline ecommerce/src/main/java/com/yetanotheruseless/industrycrashcourse/ecommerce/shoppingCart/ShoppingCartRepository.java:19]: this is awesome.  We'll go over this a little in class

File: ecommerce/src/test/java/com/yetanotheruseless/industrycrashcourse/ecommerce/EcommerceApiTests.java (hunk 1 above)
Comment (jakemannix) [reply ecommerce/src/test/java/com/yetanotheruseless/industrycrashcourse/ecommerce/EcommerceApiTests.java:25]: it helps wire in things like @MockBean and MockMvc

**Review by jakemannix — APPROVED** [review]
go for it!

**Pull request merged**